src/matrix.c

```diff
@@ -51,17 +51,14 @@ int matrix_fill(int start_x, int start_y, int end_x, int end_y, RGB *color) {
 	if (start_y > end_y)
 		return 2;
 
-	int ret = 0;
 	int x;
 	int y;
 
 	for (y=start_y; y <= end_y; y++)
 		for (x=start_x; x <= end_x; x++) {
-			ret = matrix_set(x, y, color);
-			if (ret != 0)
-				return ret;
```

Comment:
The thing is if you have an cube (or anything else) and you want to swipe over the left edge, so it disappears, the operation would fail as `start_x` is not inside the matrix, resulting in disappearing of the cube before it even moved one pixel over the edge.
Solving this problem could be done in each and every animation by checking if the value is inside the matrix, or not, or by just letting the `matrix_set` operation fail a few times inside `matrix_fill` and not resulting in more code length.
You can also apply this to the top edge. The right and bottom ones should be fine.